### ISSUE TITLE
PUT on /users/{id} should not require all fields unless creating

### ISSUE THREAD

**Issue by quan** [Joes-Minions/wayfare-api#3]
A PUT request on the `UserById` resource currently requires all fields to be present in the body. Since this includes a password and we will be implementing authorization (eventually, idk how it works on mobile yet), we should not be requiring every update to include all fields, just the ones to be updated.

For example: a PUT on /users/3 with the following body should find the user with id 3 and update their email and password, leaving all other values the same:
```json
{
    "email": "barackobama@calpoly.edu",
    "password": "m4ke4meric4g4y4g4in"
}
```

One edge case would be using PUT to create a user. If a client sends a PUT request to the /users/{id} endpoint where `id` does not currently exist, by the [definition of PUT](https://tools.ietf.org/html/draft-ietf-httpbis-p2-semantics-21#section-5.3.4), that user should be created. In this case, all fields _must_ be required.

In other words, given a PUT on /users/9001:
```
if user with id 9001 exists:
    update user with new fields if fields are valid
if user with id 9001 does not exist:
    if all fields are present:
        create user if fields are valid
    else:
        raise an error
```
